keysManagement/managedPeersHolder.go

```diff
@@ -87,28 +88,41 @@ func checkManagedPeersHolderArgs(args ArgsManagedPeersHolder) error {
 	return nil
 }
 
-func createDataMap(namedIdentities []config.NamedIdentity) (map[string]*peerInfo, error) {
+func (holder *managedPeersHolder) createDataMap(namedIdentities []config.NamedIdentity) (map[string]*peerInfo, error) {
 	dataMap := make(map[string]*peerInfo)
 
 	for _, identity := range namedIdentities {
+		index := 0
 		for _, blsKey := range identity.BLSKeys {
 			bls, err := hex.DecodeString(blsKey)
 			if err != nil {
 				return nil, fmt.Errorf("%w for key %s", ErrInvalidKey, blsKey)
 			}
+			if len(bls) == 0 {
+				continue
+			}
 
 			blsStr := string(bls)
 			dataMap[blsStr] = &peerInfo{
 				machineID:    generateRandomMachineID(),
-				nodeName:     identity.NodeName,
+				nodeName:     generateNodeName(identity.NodeName, index),
 				nodeIdentity: identity.Identity,
 			}
+			index++
```

Comment:
any reason to not use the index in the bls keys list instead of the new index? could be useful to see the gaps for wrongly configured keys

Reply:
If there is a bad bls key, it will error. The reason I've implementet this way is to exclude the possible empty keys